### PR TITLE
Game screen abstraction

### PR DESCRIPTION
**Pull request by gleblebedev**
Basic implementation of a "Game screen" abstraction. It handles game screen changes and resets mouse mode to a selected settings.

In the following updates the Game Screen going to add/remove root UI element and manage other similar states.

### CODE REVIEW

File: Source/Samples/01_HelloWorld/HelloWorld.cpp [hunk 1]
```diff
@@ -72,16 +73,14 @@ void HelloWorld::CreateText()
     helloText->SetVerticalAlignment(VA_CENTER);
 
     // Add Text instance to the UI root element
-    GetSubsystem<UI>()->GetRoot()->AddChild(helloText);
+    GetRoot()->AddChild(helloText);
```
Comment (eugeneko) [inline Source/Samples/01_HelloWorld/HelloWorld.cpp:76]: Maybe GetRoot -> GetUIRoot?

File: Source/Samples/01_HelloWorld/HelloWorld.h [hunk 2]
```diff
@@ -56,5 +56,5 @@ class HelloWorld : public Sample
     /// Subscribe to application-wide logic update events.
     void SubscribeToEvents();
     /// Handle the logic update event.
-    void HandleUpdate(StringHash eventType, VariantMap& eventData);
+    virtual void Update(float timestep) override;
```
Comment (eugeneko) [inline Source/Samples/01_HelloWorld/HelloWorld.h:59]: Urho conventions don't keep virtual when override is used, let's follow them

File: Source/Samples/04_StaticScene/StaticScene.cpp [hunk 3]
```diff
@@ -35,9 +33,12 @@
 #include <Urho3D/UI/Font.h>
 #include <Urho3D/UI/Text.h>
 #include <Urho3D/UI/UI.h>
+#include <Urho3D/Input/FreeFlyController.h>
 
 #include "StaticScene.h"
 
+#include "Urho3D/Input/FreeFlyController.h"
```
Comment (eugeneko) [inline Source/Samples/04_StaticScene/StaticScene.cpp:40]: Remove

File: Source/Samples/SamplesManager.cpp [hunk 4]
```diff
@@ -187,11 +194,12 @@ void SamplesManager::Start()
 
     // Register an object factory for our custom Rotator component so that we can create them to scene nodes
     context_->AddFactoryReflection<Rotator>();
+    context_->AddFactoryReflection<SampleSelectionScreen>();
 
     inspectorNode_ = MakeShared<Scene>(context_);
 
-    input->SetMouseMode(MM_FREE);
-    input->SetMouseVisible(true);
+    sampleSelectionScreen_ = context_->CreateObject<SampleSelectionScreen>();
```
Comment (eugeneko) [inline Source/Samples/SamplesManager.cpp:201]: Please don't use context_->CreateObject

File: Source/Samples/SamplesManager.cpp [hunk 5]
```diff
@@ -376,9 +384,13 @@ void SamplesManager::StartSample(StringHash sampleType)
     IntVector2 screenSize = graphics->GetSize();
     graphics->SetMode(Max(screenSize.x_, screenSize.y_), Min(screenSize.x_, screenSize.y_));
 #endif
-    runningSample_.DynamicCast(context_->CreateObject(sampleType));
-    if (runningSample_)
-        runningSample_->Start(commandLineArgs_);
+    SharedPtr<Sample> screen;
```
Comment (eugeneko) [inline Source/Samples/SamplesManager.cpp:387]: screen -> sample/sampleState

File: Source/Samples/SamplesManager.h [hunk 6]
```diff
@@ -73,7 +88,8 @@ class SamplesManager : public Application
     void StartSample(StringHash sampleType);
 
     ///
-    SharedPtr<Sample> runningSample_;
+    //SharedPtr<Sample> runningSample_;
```
Comment (eugeneko) [inline Source/Samples/SamplesManager.h:91]: Remove

File: Source/ThirdParty/RmlUi/Source/Core/EventDispatcher.cpp [hunk 7]
```diff
@@ -171,7 +171,8 @@ bool EventDispatcher::DispatchEvent(Element* target_element, const EventId id, c
 		return true;
 
 	// Use stable_sort so that the order of the listeners in a given element is maintained.
-	std::stable_sort(listeners.begin(), listeners.end());
+    if (!listeners.empty())
```
Comment (eugeneko) [inline Source/ThirdParty/RmlUi/Source/Core/EventDispatcher.cpp:174]: ??

File: Source/Urho3D/SystemUI/Gizmo.cpp [hunk 8]
```diff
@@ -188,7 +188,9 @@ bool Gizmo::Manipulate(const Camera* camera, Node** begin, Node** end)
                 // A workaround for ImGuizmo bug where delta matrix returns absolute scale value.
                 if (!nodeScaleStart_.contains(node))
                     nodeScaleStart_[node] = node->GetScale();
-                node->SetScale(nodeScaleStart_[node] * delta.Scale());
+                node->SetScale(currentOrigin.Scale());
```
Comment (eugeneko) [inline Source/Urho3D/SystemUI/Gizmo.cpp:191]: ??

Comment (gleblebedev) [reply Source/Urho3D/SystemUI/Gizmo.cpp:191]: This fix is an another PR. I'll roll it back.

**Pull request merged**